### ISSUE TITLE
Save directory does not get created automatically

### ISSUE THREAD

**Issue by silwol** [Aaron-SP/bds#21]
When trying to save the game for the first time on a computer, I see the following messages on the cli output, and the game is not saved:
```
file: could not load file '/home/<username>/.bds-game/save/keymap.0'
Loading game in NORMAL mode
file: could not save file '/home/<username>/.bds-game/save/state.0'
file: could not save file '/home/<username>/.bds-game/save/world.0'
file: could not save file '/home/<username>/.bds-game/save/keymap.0'
controls: Shutdown called by user
```
After manually creating the paths using an `mkdir -p ~/.bds-game/save`, everything works as expected.

**Comment by Aaron-SP** [comment]
The makefile has a target to create this directory, 'make savepath' as specified in the README.MD. The reason that this is not done automatically is because I don't want to overwrite user data, so it is up to the user to create this directory.

Also the game has no business snooping around people's home directories, so I am very against fixing this for convenience. Perhaps the README.MD needs to be better organized as that information could be hidden.

**Comment by silwol** [comment]
Hmm, I would consider myself an advanced computer user, and even though I compiled the program by myself, I didn't read the README file. I assume neither will most of the other new users.
I consider the savegame directory (if below the data path defined by XDG) the territory of the program, and would expect it to be created recursively. I assume this is true for most computer users, and I don't know any other program that behaves differently.
If you would like to keep it the way it is now, some feedback for the user would be helpful. Currently the user thinks the game got stored, I assume not many of them will take a look at the command-line output. They will be surprised to find that this was not the case when they try to load the game, and there are only empty slots. In such cases, I usually work with the [Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

**Comment by Aaron-SP** [comment]
Main problem is that in 2018, C++ **STILL** doesn't have a cross-platform solution to create directories. Maybe on program launch I can create the directory if not created.

**Comment by Aaron-SP** [comment]
Fixed in da4a95fcab2cf748c8d0ff1a8085b91c54ac1cc2
